src/read/xcoff/symbol.rs

```diff
@@ -288,9 +328,16 @@ impl<'data, 'file, Xcoff: FileHeader, R: ReadRef<'data>> ObjectSymbol<'data>
         self.symbol.is_undefined()
     }
 
+    /// Return true if the symbol is a definition of a function or data object.
     #[inline]
     fn is_definition(&self) -> bool {
-        self.symbol.is_definition()
+        let aux_csect = self
+            .symbols
+            .aux_csect(self.index.0, self.symbol.n_numaux() as usize)
+            .unwrap();
+        let smclas = aux_csect.x_smclas();
```

Comment:
From the doc: The x_smclas field is used only when the value of bits 5-7 of the x_smtyp field is either XTY_SD or XTY_CM.